ext/ffi/lib.rs

```diff
@@ -583,23 +650,71 @@ fn ffi_call(args: FfiCallArgs, symbol: &Symbol) -> Result<Value, AnyError> {
           native_values.push(value);
         }
       },
+      NativeType::Function { .. } => {
+        if let Some(idx) = value.as_u64() {
+          // Placeholder to reserve spot for the function `Arg`.
+          let native_value = NativeValue { void_value: () };
+          offsets.push(idx as usize);
+          native_values.push(native_value);
+        }
+      }
       _ => {
         let value = NativeValue::new(native_type, value)?;
         native_values.push(value);
       }
     }
   }
 
-  let call_args = symbol
+  let mut call_args = symbol
     .parameter_types
-    .iter()
+    .clone()
+    .into_iter()
     .zip(native_values.iter())
-    .map(|(&native_type, native_value)| unsafe {
+    .map(|(native_type, native_value)| unsafe {
       native_value.as_arg(native_type)
     })
     .collect::<Vec<_>>();
 
-  Ok(match symbol.result_type {
+  if !offsets.is_empty() {
+    for (cb, offset) in functions.into_iter().zip(offsets) {
+      // Grab the persistent for callback & current scope.
+      let global = ISOLATE.with(|cell| unsafe {
+        let isolate = cell.borrow_mut().as_ref().unwrap().isolate;
+        // NOTE: `into_raw` leaks.
+        let global = v8::Global::new(&mut *isolate, cb);
+        global.into_raw()
+      });
+      dbg!(global.as_ptr() as u64);
+      let info = CallbackInfo::new(global);
+      let signature = &symbol.parameter_types[offset];
+      let cif = match signature {
+        NativeType::Function {
+          ref parameters,
+          ref result,
+        } => Cif::new(
+            parameters.iter().map(libffi::middle::Type::from),
+            libffi::middle::Type::from(*result.clone()),
+          ),
+        _ => unreachable!(),
+      };
+
+      let closure =
+        libffi::middle::Closure::new(cif, deno_ffi_callback, &info);
+      call_args[offset] = Arg::new(closure.code_ptr());
+      
+      std::mem::forget(global);
+      // So this is a bad API. We need to ensure that the destructor for the closure is not run.
```

Comment:
With purely sync callbacks, you'd explicitly want `closure` and `info` to be removed once `call_symbol` has returned.

It makes a lot of sense that the `.code_ptr()` does not consume the closure, as the pointer is just a raw pointer to the C function created for the closure, and closure owns the C function. The closure must stay alive as long as the C function is in use.

For synchronous callbacks I think this should be just:
```rs
// I guess this is explicitly needed since we're working with a raw global pointer?
std::mem::forget(global);
return Ok(call_symbol(symbol, call_args));
// allow closure, info go out of scope
```
And for the more general registered callbacks it would be something like:

```rs
fn op_register_callback(
  state: OpState,
  args: ForeignFunction, // omit name
  cb: serde_v8::Value<'_>
) {
  let closure = libffi::middle::Closure::new(cif, deno_ffi_callback, info);
  let resource = RegisteredCallbackResource {
    closure,
    info,
    parameters,
    result,
  };
  Ok(state.resource_table.add::<RegisteredCallbackResource>(resource))
}

fn ffi_call(
  //...
) {
  //...
  for (native_type, value) in symbol
    .parameter_types
    .clone()
    .into_iter()
    .zip(args.parameters.into_iter())
 {
    match native_type {
      NativeType::Function { .. } => {
        if let Some(rid) = value.as_u32() {
          native_values.push(state.resource_table.get::<RegisteredCallbackResource>(rid)?); // Error should be handled already in JS land: Keep track of when a registered callback is removed and mark the removal on the instance. RID itself is of course hidden inside the instance and is only accessible through eg. a Symbol shared with the FFI call handler
        }
      }
}
```

etc.